Runtime/Scripts/Api/V1/Api/Engines/Engine/ChatCompletion/ChatCompletionsResourceV1.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Threading.Tasks;
+
+using UnityEngine;
+
+namespace OpenAi.Api.V1
+{
+
+    /// <summary>
+    /// Resource providing completions functionality. Text generation is the core function of the API. You give the API a prompt, and it generates a completion. The way you �program� the API to do a task is by simply describing the task in plain english or providing a few written examples. This simple approach works for a wide range of use cases, including summarization, translation, grammar correction, question answering, chatbots, composing emails, and much more (see the prompt library for inspiration). <see href="https://beta.openai.com/docs/examples"/>
+    /// </summary>
+    public class ChatCompletionsResourceV1 : AApiResource<OpenAiApiV1>
+    {
+        /// <inheritdoc/>
+        public override string Endpoint => "/chat/completions";
```

Comment:
Currently in the library, when someone wants to use a REST Api call associated with `<engine_id>/completions`, they call  `EnginesResource.Engine(<engine_id>).Completions`, which follows `engines/<engine_id>/completions`.

This `ChatCompletionResource` should be split into a `ChatResource` and a `ChatCompletionsResources` with the chat resource containing something along the lines of

```csharp
 public class ChatResourceV1 : AApiResource<OpenAiApiV1>
 {
    public ChatCompletionsV1 Completions => new ChatCompletionsV1(this);
 }
```

This will maintain the call pattern. `engines/<engine_id>/chat/completions` as `EnginesResource.Engine(<engine_id>).Chat.Completions`, instead of `EnginesResource.Engine(<engine_id>).ChatCompletions`

This is necessary becuase in the future chat might end up having more API endpoints following it.